pkg/controllers/provisioning/provisioner.go

```diff
@@ -141,8 +142,17 @@ func (p *Provisioner) getPods(ctx context.Context) ([]*v1.Pod, error) {
 	}
 	var pods []*v1.Pod
 	for i := range podList.Items {
-		if isProvisionable(&podList.Items[i]) {
-			pods = append(pods, &podList.Items[i])
+		pod := podList.Items[i]
+		errs := multierr.Combine(
+			validate(&pod),
+			p.volumeTopology.validatePvcs(ctx, &pod),
+		)
+		if errs != nil {
+			logging.FromContext(ctx).Errorf("Unable to batch pod %s, %s", pod.Name, errs)
```

Comment:
Noticing I didn't log the errors in the reconcile loop, but I do here.  It's certainly valuable to know that a pod wasn't scheduled due to a missing PVC, but it's not really due to Karpenter misbehaving.  And people can always describe the pending pod to see why it wasn't scheduled.